src/Models/ShallowWaterModels/shallow_water_model.jl

```diff
@@ -66,7 +102,6 @@ function ShallowWaterModel(;
                  forcing::NamedTuple = NamedTuple(),
                              closure = nothing,
                           bathymetry = nothing,
-                            solution = nothing,
```

Comment:
removing this from the constructor seems to not have an effect on the shallow water tests -- so I guess we didn't need it there, right @francispoulin?

Reply:
I guess not.  Just to be clear, this line serves the same purpose as [line 36](https://github.com/CliMA/Oceananigans.jl/blob/4411de0ca834351f1c7075a2bee973ecbf64a3f9/src/Models/IncompressibleModels/incompressible_model.jl#L36) of `IncompressibleModel` which specifies the velocities.  I thought we needed this to have the solution as part of the constructor but maybe I'm wrong. 

Can someone explain to me why we might need this in `IncompressibleModel`?

I looked at the test and see that[ line 137 ](https://github.com/CliMA/Oceananigans.jl/blob/4411de0ca834351f1c7075a2bee973ecbf64a3f9/test/test_shallow_water_models.jl#L137) does set the mass fluxes so if that doesn't have a problem then I guess we're fine.

Reply:
There could be cases where one wants to build `VelocityFields` or tracer fields before constructing the model. In that case users can pass the velocity fields in directly. `velocitices = nothing` specifies the constructor to allocate memory and build the `VelocityFields` internally.

Nothing is "needed", it's just a question of what features are supported. The kwargs are all about supporting features and model designers should choose them with intent.